api/types/container/config.go

```diff
@@ -41,29 +41,30 @@ type HealthConfig struct {
-	WorkingDir      string              // Current directory (PWD) in the command will be launched
-	Entrypoint      strslice.StrSlice   // Entrypoint to run when starting the container
-	NetworkDisabled bool                `json:",omitempty"` // Is network disabled
-	MacAddress      string              `json:",omitempty"` // Mac Address of the container
-	OnBuild         []string            // ONBUILD metadata that were defined on the image Dockerfile
-	Labels          map[string]string   // List of labels set to this container
-	StopSignal      string              `json:",omitempty"` // Signal to stop a container
-	StopTimeout     *int                `json:",omitempty"` // Timeout (in seconds) to stop a container
-	Shell           strslice.StrSlice   `json:",omitempty"` // Shell for shell-form of RUN, CMD, ENTRYPOINT
+	Hostname           string              // Hostname
+	Domainname         string              // Domainname
+	User               string              // User that will run the command(s) inside the container, also support user:group
+	AttachStdin        bool                // Attach the standard input, makes possible user interaction
+	AttachStdout       bool                // Attach the standard output
+	AttachStderr       bool                // Attach the standard error
+	ExposedPorts       nat.PortSet         `json:",omitempty"` // List of exposed ports
+	Tty                bool                // Attach standard streams to a tty, including stdin if it is not closed.
+	OpenStdin          bool                // Open stdin
+	StdinOnce          bool                // If true, close stdin after the 1 attached client disconnects.
+	Env                []string            // List of environment variable to set in the container
+	Cmd                strslice.StrSlice   // Command to run when starting the container
+	Healthcheck        *HealthConfig       `json:",omitempty"` // Healthcheck describes how to check the container is healthy
+	ArgsEscaped        bool                `json:",omitempty"` // True if command is already escaped (Windows specific)
+	Image              string              // Name of the image as it was passed by the operator (e.g. could be symbolic)
+	Volumes            map[string]struct{} // List of volumes (mounts) used for the container
+	WorkingDir         string              // Current directory (PWD) in the command will be launched
+	Entrypoint         strslice.StrSlice   // Entrypoint to run when starting the container
+	NetworkDisabled    bool                `json:",omitempty"` // Is network disabled
+	MacAddress         string              `json:",omitempty"` // Mac Address of the container
+	OnBuild            []string            // ONBUILD metadata that were defined on the image Dockerfile
+	Labels             map[string]string   // List of labels set to this container
+	RuntimeAnnotations map[string]string   // List of runtime annotations set to this container
```

Comment:
This should probably go in `HostConfig` instead of `Config` because this becomes part of the image config which does not seem desirable... we already made this mistake with `Labels` which wound up creating a lot of problems that we really couldn't fix, unfortunately.